packages/server/src/utils/xdccDatabase.ts

```diff
@@ -72,14 +72,18 @@ const adaptScriptLine = (line: string) =>
 
 const filterValidEntries = (line: string[]) => line.length >= COLUMNS_PER_FILE;
 
+const isChannelExcluded = (channelName: string): boolean => config.excludedChannels.has(channelName);
+
 export const create = async (database: DatabaseContent[]) => {
 	sqliteDb?.close();
 
 	sqliteDb = createDbInstance();
 
 	const preparedStatement: StatementSync = sqliteDb.prepare("INSERT INTO files VALUES (?, ?, ?, ?, ?, ?)");
 
-	const promises = database.map(async (channel) => {
+	const filteredDatabase = database.filter((channel) => !isChannelExcluded(channel.channelName));
```

Comment:
Don't store it in the filteredDatabase variables. Directly apply the filter and then the map.